### PR TITLE
Support alternative auth in Node.js discovery samples

### PR DESCRIPTION
**Pull request by saicheems**

### CODE REVIEW

**Comment by saicheems** [comment]
any comments?

File: src/test/java/com/google/api/codegen/testdata/discoveries/mvvm_nodejs/nodejs_translate.v2.json.baseline [hunk 1]
```diff
@@ -14,38 +14,24 @@
...
-    }
-    callback(authClient);
-  });
-}
+translate.detections.list(request, function(err, response) {
+  if (err) {
+    console.log(err);
+    return;
+  }
+
```
Comment (saicheems) [inline src/test/java/com/google/api/codegen/testdata/discoveries/mvvm_nodejs/nodejs_translate.v2.json.baseline]: Sweet! Will follow up later

**Comment by tcoffee-google** [comment]
LGTM

File: src/test/java/com/google/api/codegen/testdata/discoveries/mvvm_nodejs/nodejs_translate.v2.json.baseline [hunk 2]
```diff
@@ -14,38 +14,24 @@
...
-    if (err) {
-      console.log(err);
-      return;
-    }
-
-    // TODO: Change code below to process the `response` object:
-    console.log(JSON.stringify(response, null, 2));
-  });
-});
+var apiKey = ''; // TODO: Update placeholder with desired API key.
```
Comment (vchudnov-g) [inline src/test/java/com/google/api/codegen/testdata/discoveries/mvvm_nodejs/nodejs_translate.v2.json.baseline]: s/desired/your/ or s/desired/the appropriate/ ?
(I'm OK deferring this until a general consistency sweep)

File: src/test/java/com/google/api/codegen/testdata/discoveries/mvvm_nodejs/nodejs_translate.v2.json.baseline (hunk 1 above)
Comment (vchudnov-g) [inline src/test/java/com/google/api/codegen/testdata/discoveries/mvvm_nodejs/nodejs_translate.v2.json.baseline]: Hmm, OK if you do it for now, and then we have a review with other language experts. My own bias is to keep the whitespace (I would in my code), but perhaps for a snippet the vertical space is too costly.

File: src/test/java/com/google/api/codegen/testdata/discoveries/mvvm_nodejs/nodejs_translate.v2.json.baseline (hunk 2 above)
Comment (saicheems) [inline src/test/java/com/google/api/codegen/testdata/discoveries/mvvm_nodejs/nodejs_translate.v2.json.baseline]: @tcoffee-google ?

Comment (tcoffee-google) [inline src/test/java/com/google/api/codegen/testdata/discoveries/mvvm_nodejs/nodejs_translate.v2.json.baseline]: I like the original. Comparatively, "your" suggests that only one exists, and "the appropriate" suggests that only one is correct, whereas in reality a user may have their own reasons for choosing one or another.

Comment (tcoffee-google) [inline src/test/java/com/google/api/codegen/testdata/discoveries/mvvm_nodejs/nodejs_translate.v2.json.baseline]: That interpretation also seems reasonable to me, but I see no reason to prefer it over the more succinct "desired".

**Pull request merged**